microsetta_public_api/api/diversity/beta.py

```diff
@@ -1,6 +1,39 @@
+from microsetta_public_api.utils import jsonify
+from microsetta_public_api.resources_alt import get_resources
+from microsetta_public_api.repo._beta_repo import BetaRepo
+from microsetta_public_api.exceptions import (
+    UnknownResource,
+)
+from microsetta_public_api.config import schema
+
+
+def _validate_dataset_beta(dataset, resource_getter):
+    try:
+        dataset_resource = resource_getter().gets('datasets', dataset)
+    except KeyError:
+        raise UnknownResource(f"Unknown dataset: '{dataset}'")
```

Comment:
This seems to just be reraising, here and on line 18. Why not just have the `.gets` method raise `UnknownResource`?

Reply:
The `Element` classes doen't know what you are looking for, and I would like to keep it that way so the `Element` so it doesn't become coupled to the client code. So I can provide a more specific error message this way, since the API level knows about `Elements` and knows the meaning of what you asked it for.

Reply:
Okay, in that case I would recommend a pattern like:

```python
getter = resource_getter()
if ('datasets', dataset) not in getter:
    raise UnknownResource(...)
dataset_resource = getter.gets('datasets', dataset)
```

Or

```python
dataset_resource = resource_getter().gets('datasets', dataset, default=None)
if dataset_resource is None:
    raise UnknownResource(...)
```

...or something to that effect. The motivation is the test being done is explicit rather than implicit. The first one is perhaps more explicit. The second one corresponds a little more to what `{}.get` does

Reply:
Sure. We could make the checks explicit by using the `has` method. Code is more readable at the (admittedly low) expense of iterating over the (small) tree twice.

See https://github.com/biocore/microsetta-public-api/pull/69/commits/85c6ec8ef4b8fd0cf92aa3ff066e565f043362a4

Reply:
Thanks!!